test/jdk/javax/swing/JFileChooser/FileSystemView/SystemIconTest.java

```diff
@@ -39,20 +38,45 @@
 
     public static void main(String[] args) {
         testSystemIcon();
-        System.out.println("ok");
+        negativeTests();
     }
 
     static void testSystemIcon() {
-        String windir = System.getenv("windir");
-        testSystemIcon(new File(windir));
-        testSystemIcon(new File(windir + "/explorer.exe"));
-        return;
+        String os = System.getProperty("os.name");
+        if (os.startsWith("Windows")) {
+            String windir = System.getenv("windir");
+            testSystemIcon(new File(windir), true);
+            testSystemIcon(new File(windir + "/explorer.exe"),
+                    true);
+        } else {
+            String homedir = System.getProperty("user.home");
+            testSystemIcon(new File(homedir), false);
+        }
+    }
+
+    static void negativeTests() {
+        ImageIcon icon;
+        boolean gotException = false;
+        try {
+            icon = (ImageIcon) fsv.getSystemIcon(new File("."), -1, 16);
+        } catch (IllegalArgumentException iae) {
+            gotException = true;
+        }
+        if (!gotException) {
+            throw new RuntimeException("Negative size icon should throw invalid argument exception");
+        }
```

Comment:
A suggestion: throw the exception inside try-block and ignore the expected exception, then `gotException` can be dropped.
```suggestion
        try {
            fsv.getSystemIcon(new File("."), -1, 16);
            throw new RuntimeException("Negative size icon should throw invalid argument exception");
        } catch (IllegalArgumentException iae) {
            // expected
        }
```

Shall the test also exercise 0 as an invalid parameter? Shall the test also pass an invalid height?

Reply:
Fixed.

Reply:
I do not think such detailed testing is required. After all it is not a full spec conformance test.